src/builder/QuarkBuilder.sol

```diff
@@ -345,4 +374,25 @@ contract QuarkBuilder {
             }
         }
     }
+
+    function totalAvailableAsset(
+        string memory tokenSymbol,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) internal pure returns (uint256) {
+        uint256 total = 0;
+
+        for (uint256 i = 0; i < chainAccountsList.length; ++i) {
+            total += Accounts.sumBalances(
+                Accounts.findAssetPositions(tokenSymbol, chainAccountsList[i].chainId, chainAccountsList)
+            );
+
+            // Account for max cost if the payment token is the transfer token
+            // Simply offset the max cost from the available asset batch
+            if (payment.isToken && Strings.stringEqIgnoreCase(payment.currency, tokenSymbol)) {
+                total -= PaymentInfo.findMaxCost(payment, chainAccountsList[i].chainId);
+            }
```

Comment:
My thought was to try to make it not rely on TransferIntent as non bridgeable errors could be caught in `assertFundsAvailable` which will take in transferIntent and know which chain need to bridge or not and throw errors if needed. 

But this actually goes into a interesting UX case as user has 
```
User has: 
10 USDC on mainnet
5 USDC  on Base
4 USDC on Unbridgeable L2

User requests transfermax on mainnet. 
Do we: 
A)
Trying to transfermax on 19 USDC, then throw user errors with `FundsUnavailable()` from assertFundsAvailable

B)
We know 4USDC is unbridgable, so we adjust that so that even user requests to transfermax,  we only tranfser 15USDC and ignore 4USDC on Unbridgeable L2.  
After actions completes, user will see 15USDC transferrred out on mainnet,  but will still see 4USDC on Unbridgeable L2.
```

Reply:
Discussed offline in discord, 
Looks like we're going with A).

Reply:
Do we have a test for A? Correct me if I'm wrong, but I don't see any diffs related to the decision here.

Reply:
Oh yeah I don't expect logics diff occurs in for A,  as the follow-up assertFundAvailable will throw errors. 
But yeah I can add a test case to verify in A) case, we will get FundsUnavailable() error.

Reply:
I have added a test case to cover for A.  👍